diff --git a/sys/sys/genpercpu.c b/sys/sys/genpercpu.c
index 3769a66..76b0a68 100644
--- a/sys/sys/genpercpu.c
+++ b/sys/sys/genpercpu.c
@@ -28,6 +28,8 @@
 #define	INTSTACKSIZE	(3 * PAGESIZE)
 #endif

-#define	PCPUELT(a,t,n,e,b) { a, "t", "n", sizeof (t), e, "e", b }
+#define	PCPUELT_STR1(x)	#x
+#define	PCPUELT_STR(x)	PCPUELT_STR1(x)
+#define	PCPUELT(a,t,n,e,b) { a, PCPUELT_STR(t), PCPUELT_STR(n), sizeof (t), e, PCPUELT_STR(e), b }

 #define	ST	struct

diff --git a/sys/nfs/nfs_dump.c b/sys/nfs/nfs_dump.c
index 4c6bf42..a3f4d88 100644
--- a/sys/nfs/nfs_dump.c
+++ b/sys/nfs/nfs_dump.c
@@ -33,5 +33,7 @@
 #include <nfs/nfs_clnt.h>
 #include <nfs/rnode.h>

+static int dump_auth_marshall();
+
 #define	TIMEO		2	/* seconds before retransmitting request */
 #define	MAXTRIES	5	/* before giving up on server */
