diff --git a/sys/sys/genpercpu.c b/sys/sys/genpercpu.c
index 3769a66..76b0a68 100644
--- a/sys/sys/genpercpu.c
+++ b/sys/sys/genpercpu.c
@@ -28,6 +28,8 @@
 #define	INTSTACKSIZE	(3 * PAGESIZE)
 #endif

-#define	PCPUELT(a,t,n,e,b) { a, "t", "n", sizeof (t), e, "e", b }
+#define	PCPUELT_STR1(x)	#x
+#define	PCPUELT_STR(x)	PCPUELT_STR1(x)
+#define	PCPUELT(a,t,n,e,b) { a, PCPUELT_STR(t), PCPUELT_STR(n), sizeof (t), e, PCPUELT_STR(e), b }

 #define	ST	struct

diff --git a/sys/nfs/nfs_dump.c b/sys/nfs/nfs_dump.c
index 4c6bf42..a3f4d88 100644
--- a/sys/nfs/nfs_dump.c
+++ b/sys/nfs/nfs_dump.c
@@ -33,5 +33,7 @@
 #include <nfs/nfs_clnt.h>
 #include <nfs/rnode.h>

+static int dump_auth_marshall();
+
 #define	TIMEO		2	/* seconds before retransmitting request */
 #define	MAXTRIES	5	/* before giving up on server */

diff --git a/sys/nfs/nfs_server.c b/sys/nfs/nfs_server.c
index 3f8b5ce..0e5a6a7 100644
--- a/sys/nfs/nfs_server.c
+++ b/sys/nfs/nfs_server.c
@@ -37,6 +37,7 @@
 #include <machine/mmu.h>

 extern struct vnodeops nfs_vnodeops;
+static int mbuf_to_iov();
 /*
  * states of requests for duplicate request caching
  */

diff --git a/sys/nfs/nfs_subr.c b/sys/nfs/nfs_subr.c
index 5cc7e61..bd76079 100644
--- a/sys/nfs/nfs_subr.c
+++ b/sys/nfs/nfs_subr.c
@@ -66,5 +66,6 @@

 extern struct vnodeops nfs_vnodeops;
 static struct rnode *rfind();
+static int rp_addhash();

 extern int nrnode;		/* max rnodes to alloc, set in machdep.c */

diff --git a/sys/nfs/nfs_vfsops.c b/sys/nfs/nfs_vfsops.c
index 27bd6fa..37c1d4b 100644
--- a/sys/nfs/nfs_vfsops.c
+++ b/sys/nfs/nfs_vfsops.c
@@ -43,6 +43,13 @@
 #endif

 static int nfsrootvp();
+extern char *strcpy();
+static enum clnt_stat callrpc();
+static int getfile();
+static int mountnfs();
+static int pathconf_get();
+static int pathconf_rele();
+static int hostpath();
 extern struct vnode *makenfsnode();
 #define	MAPSIZE  256/NBBY
 static char nfs_minmap[MAPSIZE]; /* Map for minor device allocation */

diff --git a/sys/nfs/nfs_vnodeops.c b/sys/nfs/nfs_vnodeops.c
index 42e191d..a47e6b9 100644
--- a/sys/nfs/nfs_vnodeops.c
+++ b/sys/nfs/nfs_vnodeops.c
@@ -116,6 +116,10 @@
 static	int nfs_putpage();
 static	int nfs_map();
 static	int nfs_dump();
+static	int writerp();
+static	int nfswrite();
+static	int nfs_strategy();
+static	int do_bio();
 static  int nfs_cmp();
 static  int nfs_realvp();
 static  int nfs_cntl();

diff --git a/sys/os/kern_descrip.c b/sys/os/kern_descrip.c
index 7e8de7c..4f9d3bb 100644
--- a/sys/os/kern_descrip.c
+++ b/sys/os/kern_descrip.c
@@ -41,7 +41,8 @@
  */
 static void	dupit(/* fd, fp, flags */);
 static void	alloc_fd_slots();
+static struct vnode *myfhtovp();

 /*
  * System calls on descriptors.
  */
